src/parser/mod.rs

```diff
@@ -7156,6 +7156,35 @@ impl<'a> Parser<'a> {
                 partition,
                 with_name,
             }
+        } else if self.dialect.supports_alter_table_update() && self.parse_keyword(Keyword::UPDATE)
+        {
+            let mut assignments = vec![];
```

Comment:
One minor request, could we move this new block to its own function to keep the branch of the current function smaller/more-focused?